src/migrations/annotation.ts

```diff
@@ -0,0 +1,23 @@
+import { KustoQuery, defaultQuery } from 'types';
+
+export const migrateAnnotation = (annotation: any) => {
```

Comment:
I'm not enthusiastic about introducing any more untyped `any`s into the code - I think it's really easy to ship code that breaks with it, and it's not a pattern I would like to see encouraged further.

Instead is it possible to use `unknown` or `AnnotationQuery<T>`, or a specific type (with all optional fields) that forces us to make sure we're checking the types of things?

Reply:
I added a specific type since the `database` field makes it not a AnnotationQuery, and I couldn't figure out what type it actually is

Reply:
seems like a chill solution to me! Fun typescript facts just for future reference you can [extend types](https://www.typescriptlang.org/docs/handbook/2/objects.html#extending-types) if you want, or you could edit the original AnnotationQuery to have an optional database property with a question mark.

Reply:
IIRC `AnnotationQuery`  is typed at the end with `[key: string]: string` or something like that, meaning it allows any named property.